pkgs/top-level/all-packages.nix

```diff
@@ -2774,6 +2774,8 @@ let
 
   pytrainer = callPackage ../applications/misc/pytrainer { };
 
+  remarshal = (callPackage ../development/tools/remarshal { }).bin;
```

Comment:
You removed the part where bin is actually used as the output when the attr
is referenced directly. This is all a bit confusing but basically, when
referenced directly as a string, the current derivation will be used as an
output (in this case .out). However, environment.system packages and
nix-env will install all outputs for a derivation based on the outputs list
regardless of what the default is. Therefore you need both since we want to
make sure only the bin is ever referenced.
On Mon, Nov 16, 2015 at 4:41 AM Jaka Hudoklin <notifications@github.com>
wrote:

> In pkgs/top-level/all-packages.nix
> <https://github.com/NixOS/nixpkgs/pull/11048#discussion_r44918679>:
>
> > @@ -2774,6 +2774,8 @@ let
> >
> >    pytrainer = callPackage ../applications/misc/pytrainer { };
> >
> > +  remarshal = (callPackage ../development/tools/remarshal { }).bin;
>
> Ok, i did that, but it is kind of ugly in this cases bin should be default
> output.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/11048/files#r44918679>.
>